src/vcSettingsUI.cpp

```diff
@@ -21,7 +21,62 @@
 #include "udStringUtil.h"
 #include "udFile.h"
 
-#define MAX_DISPLACEMENT 1000.f
+#define MAX_DISPLACEMENT 10000.f
+
+/*
+Add errors by placing an entry in the language file with the error message, this entry name goes below in settingsErrors[].
+Add a colour in vec4 RGBA format below, and add to the error enum vcSE_ vcSettingsErrors then use vcSettingsUI_Set/Unset/Check Error functions
+*/
+static const char *settingsErrors[] =
+{
+  "bindingsErrorBound",
+  "bindingsSelectKey"
+};
+
+static const ImVec4 settingsErrorColours[] =
+{
+  ImVec4(1, 0, 0, 1),
+  ImVec4(1, 1, 1, 1)
+};
+
+static const vcSettingCategory categoryMapping[] =
+{
+  vcSC_Appearance,
+  vcSC_InputControls,
+  vcSC_Viewport,
+  vcSC_MapsElevation,
+  vcSC_Visualization,
+  vcSC_Bindings,
+  vcSC_Convert,
+  vcSC_Count,
+  vcSC_Count,
+  vcSC_Count,
+  vcSC_Count
+};
+UDCOMPILEASSERT(vcSC_Count == 10, "Update the above mapping if necessary");
+UDCOMPILEASSERT(vcSR_Count == 11, "Update the above mapping if necessary");
+
+static int errors = 0;
```

Comment:
This really should be in the `pProgramState` structure, global variables are strongly discouraged.